scripts/builtin/l2svm.dml

```diff
@@ -0,0 +1,148 @@
+#                               1 = add intercept;
+# tol       Double  0.001       Tolerance (epsilon);
+# reg       Double  1.0         Regularization parameter (lambda) for L2 regularization
+# maxiter   Int     100         Maximum number of conjugate gradient iterations
+# model     String  ---         Location to write model
+# fmt       String  "text"      The output format of the output, such as "text" or "csv"
+# Log       String  ---         [OPTIONAL] Location to write the log file
+# ---------------------------------------------------------------------------------------------
+
+
+
+
+m_l2svm = function(Matrix[Double] X, Matrix[Double] Y, Integer intercept = 0,
+Double epsilon = 0.001, Double lambda = 1, Integer maxiterations = 100, Boolean verbose = TRUE)
+    return(Matrix[Double] model)
+{
+
+#check input parameter assertions
+if(nrow(X) < 2)
+  stop("Stopping due to invalid inputs: Not possible to learn a binary class classifier without at least 2 rows")
+if(intercept != 0 & intercept != 1)
+  stop("Stopping due to invalid argument: Currently supported intercept options are 0 and 1")
+if(epsilon < 0)
+  stop("Stopping due to invalid argument: Tolerance (tol) must be non-negative")
+if(lambda < 0)
+  stop("Stopping due to invalid argument: Regularization constant (reg) must be non-negative")
+if(maxiterations < 1)
+  stop("Stopping due to invalid argument: Maximum iterations should be a positive integer")
+
+#check input lables and transform into -1/1
+check_min = min(Y)
+check_max = max(Y)
+
+num_min = sum(Y == check_min)
+num_max = sum(Y == check_max)
+
+if(num_min + num_max != nrow(Y)) print("please check Y, it should contain only 2 labels")
+else{
+	if(check_min != -1 | check_max != +1)
+		Y = 2/(check_max - check_min)*Y - (check_min + check_max)/(check_max - check_min)
+}
+
+if(verbose) print('running L2-SVM ');
+
+num_samples = nrow(X)
+dimensions = ncol(X)
+
+if (intercept == 1) {
+	ones  = matrix(1, rows=num_samples, cols=1)
+	X = cbind(X, ones);
+}
+
+num_rows_in_w = dimensions
+if(intercept == 1){
+	num_rows_in_w = num_rows_in_w + 1
+}
+w = matrix(0, rows=num_rows_in_w, cols=1)
+
+g_old = t(X) %*% Y
+s = g_old
+
+Xw = matrix(0, rows=nrow(X), cols=1)
+debug_str = "# Iter, Obj"
+iter = 0
+continue = 1
+while(continue == 1 & iter < maxiterations)  {
+	# minimizing primal obj along direction s
+	step_sz = 0
+	Xd = X %*% s
+	wd = lambda * sum(w * s)
+	dd = lambda * sum(s * s)
+	continue1 = 1
+	while(continue1 == 1){
+		tmp_Xw = Xw + step_sz*Xd
+		out = 1 - Y * (tmp_Xw)
+		sv = (out > 0)
+		out = out * sv
+		g = wd + step_sz*dd - sum(out * Y * Xd)
+		h = dd + sum(Xd * sv * Xd)
+		step_sz = step_sz - g/h
+		continue1 = (g*g/h >= epsilon)
+	}
+
+	#update weights
+	w = w + step_sz*s
+	Xw = Xw + step_sz*Xd
+
+	out = 1 - Y * Xw
+	sv = (out > 0)
+	out = sv * out
+	obj = 0.5 * sum(out * out) + lambda/2 * sum(w * w)
+	g_new = t(X) %*% (out * Y) - lambda * w
+
+	print("OBJ = " + obj)
+	debug_str = append(debug_str, iter + "," + obj)
```

Comment:
Remove the debug_str as discussed above. Also, please make the print conditional on the verbose flag (plus a more descriptive output including the iteration number like we had in the debug string before).